config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -69,17 +71,24 @@ public void upsertConfig(
   public void getConfig(
       GetConfigRequest request, StreamObserver<GetConfigResponse> responseObserver) {
     try {
-      ContextSpecificConfig config = configStore.getConfig(getConfigResourceContext(request));
-
+      ConfigResourceContext configResourceContext = getConfigResourceContext(request);
+      Optional<ContextSpecificConfig> maybeConfig = configStore.getConfig(configResourceContext);
+      // initialize empty config if not present to handle merges better
+      if (maybeConfig.isEmpty()) {
+        maybeConfig = Optional.of(emptyConfig(configResourceContext.getContext()));
+      }
       // get the configs for the contexts mentioned in the request and merge them in the specified
       // order
       for (String context : request.getContextsList()) {
-        ContextSpecificConfig contextSpecificConfig =
+        Optional<ContextSpecificConfig> contextSpecificConfig =
             configStore.getConfig(getConfigResourceContext(request, context));
-        config = merge(config, contextSpecificConfig);
+        if (contextSpecificConfig.isEmpty()) {
```

Comment:
Same thing here. If you find yourself using .get(), .isEmpty() etc. you're likely misusing an optional.
```java
config = maybeContextConfig
  .map(contextConfig -> merge(config, contextConfig))
  .orElse(config);
```

Reply:
true, but can easily copy the ref like
```
ContextSpecificConfig lastConfig = config;
config = maybeContextConfig
  .map(contextConfig -> merge(lastConfig, contextConfig))
  .orElse(config);
```

Reply:
yeah this just looks weird to me that's why